Documentation for All-In-One docker image (#1487)
dotse/zonemaster: docs/internal/maintenance/ReleaseProcess-create-docker-image.md

```sh
(cd zonemaster-backend; git clean -dfx; git reset --hard; perl Makefile.PL)
```
```sh
(cd zonemaster-gui; git clean -dfx; git reset --hard)

> ```

Reviewer: Building the GUI is tricky. To make this work I had to run the command
```
nvm install 24
```
first (even though I have done that several times). Maybe the following command would have been enough:
```
nvm use 24
```

Have you seen the issue?

Author: Yes, you are right but user can use other tools to manage Node version.

```sh
make -C zonemaster-backend all dist docker-build
```

> ```sh

Author: ```suggestion
Node 24 is required to build the GUI. If you are using `nvm` you can set it with `nvm install 24`.
```sh
```

Reviewer 2: Or we could just link to https://github.com/zonemaster/zonemaster/blob/master/docs/internal/distrib-testing/Ubuntu-Node.js-build-environment.md instead, same as in e.g. https://github.com/zonemaster/zonemaster/blob/master/docs/internal/maintenance/ReleaseProcess-create-test-distribution.md?plain=1#L108-L129.

make -C zonemaster-backend all dist docker-build
```

Node.js version 24 is required to build the GUI. If you are using `nvm` you can set it with `nvm install 24`.
If you get building errors, repeat the `nvm` commands in [build environment for Node.js] first.

```sh

> cd zonemaster-gui; npm install; npm run release; make docker-build; cd ..

Reviewer 3: Running those commands in a subshell is a little bit cleaner:
```suggestion
(cd zonemaster-gui; npm install && npm run release && make docker-build)
```

make -C zonemaster-backend all dist docker-build
```

Node 24 is required to build the GUI. If you are using `nvm` you can set it with `nvm install 24`.
You will find more information on Node.js 

> If you get building errors, repeat the `nvm` commands in [build environment for Node.js] first.

Reviewer 2: ```suggestion
Node.js version 24 is required to build the GUI. If you are using `nvm` you can set it with `nvm install 24`.
If you get building errors, repeat the `nvm` commands in [build environment for Node.js] first.
```

docker images
```

### Tag the Zonemaster-All-In-One image

For the Zonemaster-All-In-One image, add a version tag and a tag "latest".

* Add version tag:
```sh
make -C zonemaster-gui docker-tag-version
```

* Add tag "latest":
```sh
make -C zonemaster-gui docker-tag-latest
```

* If "dash version" is to be used, set tag with that version and remove tag with
plain version where "v0.0.0" should be the local version and "v0.0.0-N" should be
the "dash version" determined above:
```
cd zonemaster-gui
docker tag zonemaster/all-in-one:local zonemaster/all-in-one:v0.0.0-N
docker rmi zonemaster/all-in-one:0.0.0
cd ..
```

All the created images can now be listed. Also consider doing [sanity checks] to
verify that all images work. Images without tag are temporary images without

> further use. List images:

Reviewer 2: ```suggestion
further use. To list images:
```

And run `docker ps -a` to ensure that the backend is no longer running.

### Zonemaster-All-In-One

Start the GUI in the background:

```

> docker run --rm -d --name zmaio -p 8080:80 zonemaster/all-in-one:local gui

Reviewer 2: ```suggestion
docker run --rm -d --name zm-aio -p 8080:80 zonemaster/all-in-one:local gui
```

```

> Run `docker ps -a`. The container named `zmaio` should be running.

Reviewer 2: ```suggestion
Run `docker ps -a`. The container named `zm-aio` should be running.
```

Check if the GUI is accessible from your browser at `http://localhost:8080/` and run a test.

Finally, stop the GUI:
```sh

> docker stop zmaio

Reviewer 2: ```suggestion
docker stop zm-aio
```
